src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -233,9 +234,13 @@ internal bool TryRead(ref Utf8JsonReader reader, Type typeToConvert, JsonSeriali
                         ref reader);
                 }
 
+                populatedValue = default;
                 return true;
             }
 
+            JsonPropertyInfo? propertyInfo = state.Current.JsonPropertyInfo;
```

Comment:
We need to store property info and parent object before the `state.Push();`, I will move this down a bit below polymorphic check though because this combination is blocked anyway - we could possibly optimize that once we have Populate directly on converter